### PR TITLE
WIP: add elasticsearch callback

### PR DESCRIPTION
**Pull request by CJ-Wright**
Still needs tests and a way to setup elastic on travis-ci

### CODE REVIEW

**Comment by tacaswell** [comment]
https://docs.travis-ci.com/user/database-setup/#elasticsearch <- how to turn elastic on.

File: setup.py
Comment (pavoljuhas) [inline setup.py]: Is there any particular reason to use `0.3.6`?
I feel a brand new project should start at `0.0.1`.

Comment (CJ-Wright) [reply setup.py]: I copy pasted this from another project. Before we run a release I'd start with `0.0.0`.

**Comment by pavoljuhas** [comment]
I propose we merge this PR after adding my small correction below.
We can then open new PRs for #2 and for adding a travis test.

@stuartcampbell -- can I please have a push access for this repo?

----

@CJ-Wright -- can you please do a fast-forward of the following?

The following changes since commit 1144cfb3f2a368464e66ccada9fda678da08584f:

  add callback and setup.py (2018-10-01 13:02:06 -0400)

are available in the Git repository at:

  https://github.com/pavoljuhas/databroker-elasticsearch.git
  in branch es-01

for you to fetch changes up to 3078564283b3292ff4676ca1a9a47d3e2575d904:

  REL: reset initial version to 0.0.1 (2018-10-15 12:07:12 -0400)

----------------------------------------------------------------
Pavol Juhas (1):
      REL: reset initial version to 0.0.1

 db_es/__init__.py | 0
 setup.py          | 2 +-
 2 files changed, 1 insertion(+), 1 deletion(-)
 create mode 100644 db_es/__init__.py

**Comment by CJ-Wright** [comment]
This PR doesn't have tests or a CI running the tests, I do not want to merge this until those two things are done.

**Comment by CJ-Wright** [comment]
@stuartcampbell I also need permissions on this.

**Comment by stuartcampbell** [comment]
@CJ-Wright looks like you are already an admin, you just need to accept the invite

**Comment by CJ-Wright** [comment]
Hmm, ok that didn't show up on my github page. I've got it now.

**Comment by CJ-Wright** [comment]
@pavoljuhas does Elasticsearch automatically make a new index if we ask for an index and it doesn't exist or do we need to initialize it?

**Comment by pavoljuhas** [comment]
`db_es/__init__.py` is still missing therefore `python setup.py install` does not work.

Please add an empty `db_es/__init__.py`.

**Comment by CJ-Wright** [comment]
cycle for ci

**Comment by CJ-Wright** [comment]
@pavoljuhas how does the python elasticsearch API work?

**Comment by CJ-Wright** [comment]
Cycle for ci

**Comment by CJ-Wright** [comment]
I think this is ready to go, there are more things to do for this `sphinx/gh_pages/doctr/rever/packaging` but I don't want to have this PR wait on those as much since they are slightly separate concerns. We have baseline level functionality and tests for that functionality, with docstrings and two examples for the common usecases.

**Comment by pavoljuhas** [comment]
Please fix these whitespace issues and we can merge.

```patch
$ git diff --check master HEAD
db_es/callback.py:74: trailing whitespace.
+    
db_es/callback.py:77: trailing whitespace.
+   
db_es/callback.py:88: trailing whitespace.
+    
db_es/callback.py:95: trailing whitespace.
+    
db_es/callback.py:105: trailing whitespace.
+    
```

**Review by pavoljuhas — CHANGES_REQUESTED** [review]
Nope.  Please make sure `git diff --check master` passes without complaints.

**Review by pavoljuhas — APPROVED** [review]

**Comment by CJ-Wright** [comment]
Feel free to merge when ready.

**Pull request merged**
